### PR TITLE
Ticker abstract class v1 + CMC ticker

### PR DESCRIPTION
**Pull request by vdegenne**

### CODE REVIEW

File: packages/api/src/pubsub.ts [hunk 1]
```diff
@@ -14,7 +14,9 @@ export class PubSub {
 		pubsub.subscribe(route, callback)
 	}
 
+	// TODO: more type inconsistencies coming from your lib glenn, fix it >.<
```
Comment (VandeurenGlenn) [inline packages/api/src/pubsub.ts:17]: What is inconsistent?

Comment (vdegenne) [reply packages/api/src/pubsub.ts:17]: image

Why not just `any`

Comment (VandeurenGlenn) [reply packages/api/src/pubsub.ts:17]: That's weird, I was expecting it to be any yes.

Comment (vdegenne) [reply packages/api/src/pubsub.ts:17]: easy fix though

Comment (VandeurenGlenn) [reply packages/api/src/pubsub.ts:17]: https://github.com/VandeurenGlenn/little-pubsub/commit/5067e4e022c44f05b57fd1c4474316692edc34ae fixes this

File: packages/server/src/index.ts [hunk 2]
```diff
@@ -1,31 +1,24 @@
...
+import ms from 'ms'
 import {HttpApiServer} from './http/index.js'
 import {WSApiServer} from './ws/index.js'
 
-import {init, updateCacheWithRemote, cache} from '@blueserver/api/cache'
-import {PubSub} from '@blueserver/api/pubsub'
+/**
+ * Some inits
+ */
+// TODO: This probably needs a better place
```
Comment (VandeurenGlenn) [inline packages/server/src/index.ts:11]: looks fine by me.

File: packages/server/src/index.ts [hunk 3]
```diff
@@ -15,7 +15,7 @@ await createCacheDirectory()
  * Tickers
  */
 cmcTicker.startTicker(ms('1h'), true)
-githubTicker.startTicker(ms('5s'), true)
+githubTicker.startTicker(ms('5m'), true)
```
Comment (VandeurenGlenn) [inline packages/server/src/index.ts:18]: Haha, yes 5sec won't be good no, but every 1min should be fine? Can set to 1 if ya want.

File: packages/server/src/ws/index.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
+import {API} from '@blueserver/api'
```
Comment (VandeurenGlenn) [inline packages/server/src/ws/index.ts:1]: FInally api has an actual api 🥳

File: packages/server/src/ws/index.ts [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import {protocol} from '@blueserver/api/constants'
 import socketRequestServer from 'socket-request-server'
 import type {SocketRequestConnection} from 'socket-request-server/connection'
-import {change1h, change24h, top100} from '@blueserver/api/server/shared'
+import {change1h, change24h, top100} from '../../../api/lib/server/api.js'
```
Comment (VandeurenGlenn) [inline packages/server/src/ws/index.ts]: Auto resolved import

File: packages/api/src/api/coinmarketcap.ts [hunk 6]
```diff
@@ -1,35 +1,137 @@
...
+	async fetchListingsInfo(listingIds: number[] | number | undefined) {
+		if (listingIds === undefined) {
+			listingIds = this.getListingsInfoMap().map((l) => l.id)
+		}
+		if (!Array.isArray(listingIds)) {
+			listingIds = [listingIds]
+		}
+		logger.log('Fetching remote listings info.')
+		const past = Date.now()
+		this.#fetchListingsInfoPromise = fetch(
```
Comment (VandeurenGlenn) [inline packages/api/src/api/coinmarketcap.ts:86]: Isn't there a new thingy in Promise to handle these?

File: packages/server/src/index.ts (hunk 3 above)
Comment (vdegenne) [reply packages/server/src/index.ts:18]: for now let's just leave it until front gets stable again

File: packages/server/src/ws/index.ts (hunk 5 above)
Comment (vdegenne) [reply packages/server/src/ws/index.ts]: fixed it already no?

Comment (VandeurenGlenn) [reply packages/server/src/ws/index.ts]: Prob had tab to long open without refreshing

**Review by VandeurenGlenn — APPROVED** [review]
Lot of changes, gj!

File: packages/server/src/index.ts (hunk 3 above)
Comment (VandeurenGlenn) [reply packages/server/src/index.ts:17]: I'll create an issue for it so we don't forget.

**Pull request merged**